Assets/Scripts/GuiOutput.cs

```diff
@@ -0,0 +1,92 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+using System.Collections.Generic;
+
+public class GuiOutput : MonoBehaviour {
+
+    static private Text m_infoMessage;
+    static private Text m_contextMessage;
+    static private Text m_debugDistanceMessage;
+
+    public readonly float MAX_DELTA = 5.0F;
+    static private float elapsed;
+
+    static private Queue<string> m_messages;
+
+    private void Start()
+    {
+        m_debugDistanceMessage = GameObject.FindGameObjectWithTag("DebugDistance").GetComponent<Text>() as Text;
+        m_infoMessage = GameObject.FindGameObjectWithTag("InfoMessage").GetComponent<Text>() as Text;
+        m_contextMessage = GameObject.FindGameObjectWithTag("ContextMessage").GetComponent<Text>() as Text;
+        m_messages = new Queue<string>();

```

Comment:
I think this will search the whole scene for an active object with that tag. I found this reference that suggests a way to find a child gameobject:
http://answers.unity3d.com/questions/41528/find-a-child-with-a-name-how-to.html

So `GameObject.FindGameObjectWithTag("DebugDistance")` would become `transform.Find("DistanceDebug").gameObject`. Which is equally hard to read but is more precise since you have control over everything in the GUI prefab.